core/src/main/java/org/apache/iceberg/RollbackToSnapshot.java

```diff
@@ -19,61 +19,21 @@
 
 package org.apache.iceberg;
 
-import com.google.common.base.Preconditions;
-import org.apache.iceberg.exceptions.ValidationException;
-
-class RollbackToSnapshot implements Rollback {
-  private final TableOperations ops;
+class RollbackToSnapshot extends SnapshotManager implements Rollback {
   private TableMetadata base = null;
   private Long targetSnapshotId = null;
 
   RollbackToSnapshot(TableOperations ops) {
-    this.ops = ops;
-    this.base = ops.current(); // do not retry
+    super(ops);
   }
 
   @Override
   public Rollback toSnapshotId(long snapshotId) {
-    Preconditions.checkArgument(base.snapshot(snapshotId) != null,
-        "Cannot roll back to unknown snapshot id: %s", snapshotId);
-
-    this.targetSnapshotId = snapshotId;
-
-    return this;
+    return (Rollback) super.setCurrentSnapshot(snapshotId);
```

Comment:
I don't think the cast is needed. It can return `this` instead:

```java
public Rollback toSnapshotId(long snapshotId) {
  super.setCurrentSnapshot(snapshotId);
  return this;
}
```